fastruby/points
# [ROAD 534] Add state to stories

### Jira Ticket 
[ROAD-534](https://ombulabs.atlassian.net/browse/ROAD-534)

### Motivation / Context
Fixes #196

This PR adds a new attribute to the story model, called `approved`, which is a boolean and depending on wether it is `true`, `false` or `nil` it represents 3 possible states to a story: Approved, Rejected or Pending.

By default, all stories are created in the Pending state.

It is possible to define this state during story creation and edition.

Additionally, in the story show page, one can use a dropdown to change the state.

Finally, added labels in the project show page so we know at a glance which stories have been approved or not.
    
### QA / Testing Instructions
Run the application via `rails s` or docker and visit any project page containing stories in the stories list. You should see they are all pending.

To change the state, click on any story and then click on the dropdown button near the story title and select one of the states in the dropdown.

Additionally, you can click to edit the story and select the state from a select box or, in the project's show page, if you click to add a new story, you should be able to select it's state via a select box as well.

### Screenshots:
![image](https://github.com/fastruby/points/assets/14188887/714c28a8-42d8-4fec-96c1-6aaae5447b56)
![image](https://github.com/fastruby/points/assets/14188887/dd5ea3c4-ae62-4825-b497-8818e811cc87)
![image](https://github.com/fastruby/points/assets/14188887/e2a3a7ad-7fc0-4799-9f2c-0c5c3796861c)
![image](https://github.com/fastruby/points/assets/14188887/fa5a547a-01c2-44ac-8e91-5b473049dea9)
![image](https://github.com/fastruby/points/assets/14188887/f55babd1-b2b9-4751-86e4-7c02ae7ed927)
![image](https://github.com/fastruby/points/assets/14188887/daa2e7de-3581-4383-96d1-43da5124298c)

___

I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md).


[ROAD-534]: https://ombulabs.atlassian.net/browse/ROAD-534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
